### DISCUSSION TITLE
mock.addInteraction () not working typescript

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#220] [Q&A] [answered]
Hi , 

I am trying to convert my JS test to TS tests.
In JS I had  some code like this 
```ts 
mock.addInteraction({
            request: {
                method: 'GET',
                path: '/api/largeData'
            },
            response: {
                status: 200,
                body: mockResponseBody
            }
        });
```

However, when I use the same in TS, it's throwing me error as below .

> No overload matches this call.
>   The last overload gave the following error.
>     Argument of type '{ request: { method: string; path: string; }; response: { status: number; body: string; }; }' is not assignable to parameter of type 'Handler[]'.
>       Object literal may only specify known properties, and 'request' does not exist in type 'Handler[]'.ts(2769)
> mock.d.ts(97, 17): The last overload is declared here.

I am not sure why the overload function not working in TS.
Can someone provide some suggestion ?

Thanks,
Mihir

**Comment by ASaiAnudeep** [comment, marked as answer]
This is fixed in the latest version of pactum.

**Reply by mihirsasmal** [reply]
Thanks AsaiAnudeep. It is fixed in the version 3.2.3.
